### PR TITLE
Check node and broker id equality in KRaft mode on start up and not in broker.id setter

### PR DESCRIPTION
**Pull request by fedinskiy**
Fixes https://github.com/strimzi/test-container/issues/124

### CODE REVIEW

File: src/test/java/io/strimzi/test/container/StrimziKafkaKraftContainerIT.java [hunk 1]
```diff
@@ -128,6 +130,24 @@ void testUnsupportedKRaftUsingImageName() {
         assertThrows(UnsupportedKraftKafkaVersionException.class, () -> systemUnderTest.start());
     }
 
+    @Test
+    void testUNsupportedKraftAndIdsMismatch() {
+        systemUnderTest = new StrimziKafkaContainer("quay.io/strimzi-test-container/test-container:latest-kafka-2.8.2")
+                .withBrokerId(1)
+                .withNodeId(1)
```
Comment (see-quick) [inline src/test/java/io/strimzi/test/container/StrimziKafkaKraftContainerIT.java:137]: and I think you have to have different IDs to achieve such an exception, right?

**Comment by see-quick** [comment]
Also, there are a few checkstyle errors i.e.,
```
[ERROR] /home/vsts/work/1/s/src/test/java/io/strimzi/test/container/StrimziKafkaKraftContainerIT.java:139:35: Empty statement. [EmptyStatement]
[ERROR] /home/vsts/work/1/s/src/test/java/io/strimzi/test/container/StrimziKafkaKraftContainerIT.java:141: 'lambda arguments' has incorrect indentation level 16, expected level should be 12. [Indentation]
```
before you push always try to execute `mvn clean install -DskipTests`.

**Comment by see-quick** [comment]
Also as you removed this one test case the mutation testing is failing because the average now of what is covered is less than it was before. 
```java
Line coverage of 71(326/459) is below threshold of 72
```
So please update it also in the `pom.xml`. Specifically here [1].

[1] - https://github.com/strimzi/test-container/blob/main/pom.xml#L458

**Review by see-quick — APPROVED** [review]
Thanks

**Review by scholzj — APPROVED** [review]

**Pull request merged**